cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/BasePagingService.cs

```diff
@@ -25,10 +34,13 @@ public virtual async Task<Stream> GetPagedDataAsync(Func<RequestInformation, Can
         Stream? response = null;
         do
         {
-            var pageData = await requestExecutorAsync(requestInfo, cancellationToken);
+            var responseHandler = CreateResponseHandler();
+            await requestExecutorAsync(requestInfo, responseHandler, cancellationToken);
```

Comment:
Passing a `IResponseHandler` means that the error handling by throwing the appropriate `ApiException` for non-success status codes (4XX and 5XX) is lost if the ResponseHandler does not do this (as in the case for `NativeResponseHandler`). Is this the desired result for the service?

Reply:
We want access to the Content-Type header. At the moment, this is the only viable way to get that.